### PR TITLE
Optimize the granularity of synchronization locks.

### PR DESCRIPTION
**Pull request by iteyelmp**

### CODE REVIEW

File: src/utils/uploader.js [hunk 1]
```diff
@@ -112,18 +112,35 @@ export class BlobUploader {
...
     }
 
-    async sendTxLock(tx, blobs) {
+    async sendTx(tx, blobs) {
+        const txObj = await this.#populate(tx, blobs);
+        return await this.#wallet.sendTransaction(txObj);
+    }
+
+    async #sendTxLock(tx) {
         const release = await this.#mutex.acquire();
```
Comment (syntrust) [inline src/utils/uploader.js:124]: Which part does mutex protect?

File: src/utils/uploader.js [hunk 2]
```diff
@@ -112,18 +112,35 @@ export class BlobUploader {
...
         try {
-            return await this.sendTx(tx, blobs);
+            tx.nonce = await this.getNonce();
+            return await this.#wallet.provider.broadcastTransaction(await this.#wallet.signTransaction(tx));
```
Comment (syntrust) [inline src/utils/uploader.js:127]: What is the difference between this line and wallet.sendTransaction?

File: src/utils/uploader.js [hunk 3]
```diff
@@ -112,18 +112,35 @@ export class BlobUploader {
         tx.blobVersionedHashes = versionedHashes;
         tx.blobs = ethersBlobs;
         tx.kzg = kzg;
-        return await this.#wallet.sendTransaction(tx);
+        return tx;
     }
 
-    async sendTxLock(tx, blobs) {
+    async sendTx(tx, blobs) {
```
Comment (syntrust) [inline src/utils/uploader.js:118]: What is the use case of sendTx instead of  sendTxLock?

File: src/utils/uploader.js (hunk 2 above)
Comment (iteyelmp) [reply src/utils/uploader.js:127]: There are many network requests inside ‘sendTransaction’

File: src/utils/uploader.js (hunk 3 above)
Comment (iteyelmp) [reply src/utils/uploader.js:118]: Synchronization locks will only be executed for concurrent transactions, and EthStorage ‘write’ are not concurrent.

File: src/utils/uploader.js (hunk 1 above)
Comment (iteyelmp) [reply src/utils/uploader.js:124]: Ensure that the acquisition of nonce and the broadcast of tx are incremented in order.

File: src/utils/uploader.js [hunk 4]
```diff
@@ -112,18 +112,35 @@ export class BlobUploader {
...
             release();
         }
     }
 
+    async sendTxLock(tx, blobs) {
+        const txObj = await this.#populate(tx, blobs);
+
+        // init
+        txObj.nonce = 0;
+        const pop = await this.#wallet.populateTransaction(tx);
```
Comment (qzhodl) [inline src/utils/uploader.js:138]: It seems to be a little over-complicated

**Pull request closed without merging**